curvefs/src/client/fuse_client.cpp

```diff
@@ -638,55 +648,79 @@ CURVEFS_ERROR FuseClient::FuseOpRename(fuse_req_t req, fuse_ino_t parent,
     RETURN_IF_UNSUCCESS(CommitTx);
     renameOp.UnlinkSrcParentInode();
     renameOp.UnlinkOldInode();
+    renameOp.UpdateInodeParent();
     renameOp.UpdateCache();
 
-    if (enableSumInDir_ && parent != newparent) {
-        // update parent relationship
-        Dentry dentry;
-        rc = dentryManager_->GetDentry(newparent, newname, &dentry);
-        if (rc != CURVEFS_ERROR::OK) {
-            LOG(WARNING) << "dentryManager_ GetDentry fail, ret = " << rc
-                        << ", parent = " << newparent << ", name = " << newname;
-            return rc;
-        }
-        uint64_t ino = dentry.inodeid();
-
-        std::shared_ptr<InodeWrapper> inodeWrapper;
-        rc = inodeManager_->GetInode(ino, inodeWrapper);
-        if (rc != CURVEFS_ERROR::OK) {
-            LOG(ERROR) << "inodeManager get inode fail, ret = " << rc
-                    << ", inodeid = " << ino;
-            return rc;
-        }
-        XAttr xattr;
-        xattr.mutable_xattrinfos()->insert({XATTRENTRIES, "1"});
-        if (dentry.type() == FsFileType::TYPE_DIRECTORY) {
-            xattr.mutable_xattrinfos()->insert({XATTRSUBDIRS, "1"});
-        } else {
-            xattr.mutable_xattrinfos()->insert({XATTRFILES, "1"});
-        }
-        xattr.mutable_xattrinfos()->insert({XATTRFBYTES,
-            std::to_string(inodeWrapper->GetLength())});
+    // return ok even update xattr failed
+    CURVEFS_ERROR trc = CURVEFS_ERROR::OK;
+    if (enableSumInDir_) {
```

Comment:
maybe move the below logic into one funciton is beeter, it's too much.

Reply:
fixed